src/modules/sensors/vehicle_air_data/VehicleAirData.cpp

```diff
@@ -172,6 +172,9 @@ void VehicleAirData::Run()
 				if (_calibration[uorb_index].device_id() != report.device_id) {
 					_calibration[uorb_index].set_device_id(report.device_id);
 					_priority[uorb_index] = _calibration[uorb_index].priority();
+					calibration::SetCalibrationParam(_calibration[uorb_index].SensorString(), "ID", uorb_index, report.device_id);
```

Comment:
I agree with the intent, but you should go through calibration::Barometer rather than doing it directly.

```suggestion
					_calibration[uorb_index].ParametersSave(uorb_index);
```

Reply:
Also maybe only if not calibrated to avoid unnecessary param calls (potential locking).